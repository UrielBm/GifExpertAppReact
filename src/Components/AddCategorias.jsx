import React, {useState } from 'react'
import PropTypes from 'prop-types';
export const AddCategorias = ({setCategorias}) => {
    const [InputValue, setInputValue] = useState('');
    const OnChangeValue= (e) =>{
        setInputValue(e.target.value);
        console.log('Onchange llamado')
    };
    const handleSumit= (e)=>{
        e.preventDefault();
        if(InputValue.trim().length>2){
        setCategorias(categ=>[InputValue,...categ,]);
        setInputValue('')
        };
    };
    return (
        <form className="wrapper-form" onSubmit={handleSumit}>
            <p>{InputValue}</p>
            <label>
                <input type="text"  className="input-categoria" placeholder='"ingresa tu  nueva categoria"' value={InputValue} onChange={OnChangeValue} /> 
            </label>
        </form>
    );
}
AddCategorias.propTypes = {
    setCategorias: PropTypes.func.isRequired
};
