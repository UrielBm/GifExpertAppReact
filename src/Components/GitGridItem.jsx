import React from 'react'
import PropTypes from 'prop-types';
export const GitGridItem = ({id,title,url}) => {
    return (
        <div className="wrapper-gift-item">
            {/* {imagenes.title} */}
            < img  className="gift" src={url} alt={title} />
            <p className="titulo-gift animate__animated animate__bounceIn animate__delay-1s ">{title}</p>
        </div>
    )
}
GitGridItem.propTypes={
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}