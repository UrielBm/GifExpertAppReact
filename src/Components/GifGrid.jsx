import React, {Fragment}from 'react'
import { useFectchGifts } from '../hooks/useFectchGifs'
 import { GitGridItem } from './GitGridItem';
export const GifGrid = ({categoria}) => {
     const {loading,data:imagenes}=useFectchGifts(categoria);
    // console.log(data);

    return (
        <Fragment>
            <h3 className="titulo-Categoria">{categoria}</h3>
            {loading&& <p className="animate__animated animate__flash">Cargando...</p>}
            <div className="wrapper-card-gift-item animate__animated animate__bounceInDown">
                {
                    imagenes.map((imagenes)=>(
                        <GitGridItem key={imagenes.id} {...imagenes} />
                    ))
                }
                
            </div>
        </Fragment>
    )
}
