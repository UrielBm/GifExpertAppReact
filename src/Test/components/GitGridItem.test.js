import React from 'react';
import {shallow} from 'enzyme';
import {GitGridItem} from '../../Components/GitGridItem';
describe('Probando el uso correcto del componmente de React GitGridItem.js', () => {
        const title ="Titulo de imagen";
        const url="http://localhost/algo.jpg";
        const wrapper=shallow(<GitGridItem title={title} url={url} />);
    test('Debe mostrar el componente correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

test('Debe de tener n pÃ¡rrafo con el titulo', () => {
    const p=wrapper.find('p');
    expect(p.text().trim()).toBe(title);
    
});
test('debe mostrar bien los argumentos de la imagen ', () => {
    const img=wrapper.find('img');
    // console.log(img.props().src);
    expect(img.props().src).toBe(url);
    expect(img.prop('alt')).toBe(title);
});
test('Evaluar que las clases en el div esten agregadas ', () => {
    
    const div=wrapper.find('div');
    // console.log(div.prop('className'));
    expect(div.prop('className')).toBe('wrapper-gift-item');
})



    
})
