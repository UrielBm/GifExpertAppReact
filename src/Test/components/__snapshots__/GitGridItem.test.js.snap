// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando el uso correcto del componmente de React GitGridItem.js Debe mostrar el componente correctamente 1`] = `
<div
  className="wrapper-gift-item"
>
  <img
    alt="Titulo de imagen"
    className="gift"
    src="http://localhost/algo.jpg"
  />
  <p
    className="titulo-gift animate__animated animate__bounceIn animate__delay-1s "
  >
    Titulo de imagen
  </p>
</div>
`;
